src/contract.rs

```diff
@@ -407,20 +416,24 @@ pub fn update_stream(
         }
     }
 
-    stream.last_updated = if now < stream.start_time {
-        stream.start_time
+    stream.last_updated_block = if now_block < stream.start_block {
+        stream.start_block
     } else {
-        now
+        now_block
     };
 
     Ok((diff, new_distribution_balance))
 }
 
-fn calculate_diff(end_time: Timestamp, last_updated: Timestamp, now: Timestamp) -> Decimal {
+fn calculate_diff(end_block: u64, last_updated_block: u64, now_block: u64) -> Decimal {
     // diff = (now - last_updated) / (end_time - last_updated)
```

Comment:
let's update the comment as well

Reply:
Ok. I will update every comment related to new feature.